akka-actor/src/main/scala/akka/io/Tcp.scala

```diff
@@ -425,7 +425,7 @@ object Tcp extends ExtensionId[TcpExt] with ExtensionIdProvider {
    * Whenever a command cannot be completed, the queried actor will reply with
    * this message, wrapping the original command which failed.
    */
-  case class CommandFailed(cmd: Command) extends Event
+  case class CommandFailed(cmd: Command, cause: Option[Throwable] = None) extends Event
```

Comment:
On a related note: Akka 2.4.x is supposed to be as binary compatible to 2.3.x as we can make it, so I’m unsure about how to proceed here.